packages/itwin/measure-tools/src/measurements/AreaMeasurement.ts

```diff
@@ -298,10 +301,19 @@ export class AreaMeasurement extends Measurement {
     context.addDecorationFromBuilder(builder);
   }
 
+  public override onDrawingMetadataChanged(): void {
+    this.polygon.worldScale = this.worldScale;
+    this._polygon.recomputeFromPoints();
+  }
+
   public override decorate(context: DecorateContext): void {
     super.decorate(context);
 
-    if (this.polygonPoints.length === 0) return;
+    if (!this.viewTarget.isViewportCompatible(context.viewport))
```

Comment:
No need to this within any Measurement class. Caller will take care of it before calling Measurement.decorate()